artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -207,8 +208,8 @@ private void checkIDSupplier(NodeStore<MessageReference> nodeStore) {
    // The quantity of pagedReferences on messageReferences priority list
    private final AtomicInteger pagedReferences = new AtomicInteger(0);
 
-   // The estimate of memory being consumed by this queue. Used to calculate instances of messages to depage
-   final AtomicInteger queueMemorySize = new AtomicInteger(0);
+
+   final SizeAwareMetric queueMemorySize = new SizeAwareMetric();
```

Comment:
Ah so it did use a counter before.

Reply:
@gemmellr I preferred to use the SizeAwareMetric here instead of introducing a new counter.